curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-destructuring-assignment-with-the-rest-operator-to-reassign-array-elements.english.md

```diff
@@ -24,11 +24,11 @@ Use destructuring assignment with the rest operator to perform an effective <cod
 ```yml
 tests:
   - text: <code>arr</code> should be <code>[3,4,5,6,7,8,9,10]</code>
-    testString: assert(arr.every((v, i) => v === i + 3) && arr.length === 8,'<code>arr</code> should be <code>[3,4,5,6,7,8,9,10]</code>');
+    testString: assert(arr.every((v, i) => v === i + 3) && arr.length === 8);
   - text: <code>Array.slice()</code> should not be used.
-    testString: getUserInput => assert(!getUserInput('index').match(/slice/g), '<code>Array.slice()</code> should not be used.');
+    testString: getUserInput => assert(!getUserInput('index').match(/slice/g));
   - text: Destructuring on <code>list</code> should be used.
-    testString: getUserInput => assert(getUserInput('index').match(/\s*\[\s*,\s*,\s*\.\.\.\s*arr\s*\]\s*=\s*list\s*/g), 'Destructuring on <code>list</code> should be used.');
+    testString: assert(code.replace(/\s/g, '').match(/\[(([a-z]\w*)?,){1,}\.\.\.arr\]=list/gi));
```

Comment:
```suggestion
    testString: assert(code.replace(/\s/g, '').match(/\[(\w*,){1,}\.\.\.arr\]=list/i));
```
How about this simplification?  This way we can let the editor worry about the variable names.

Reply:
I originally did that, but thought it would not hurt to have it check it in the regex.  Is there an edge case my regex misses?

Reply:
The only ones I can think of is
```js
const [_, , ...arr] = list;
```
Well, that, `$` and unicode.  To be honest, I think your original regex was good.  I just wanted to see if you'd considered this and clearly you had!

Reply:
The `underscore` and `$`, and unicode are good points.  I am less concerned about the `$` and unicode characters, but I use the `underscore` many times in code.

Maybe we can account for the first two and not worry about the unicode for now with:
```
testString: assert(code.replace(/\s/g, '').match(/\[(([_$a-z]\w*)?,){1,}\.\.\.arr\]=list/i));
```

Reply:
@RandellDawson That makes sense.  I don't think unicode will be a problem.

Reply:
@ojeytonwilliams I have updated the `testString` with what I show above.  Thanks for your review.

Reply:
@RandellDawson No problem - it looks good.  I'll merge this when Travis has finished.